### ISSUE TITLE
Bug: Can't use highlight groups in `diagnostics_color` for background, foreground...

### ISSUE THREAD

**Issue by rafamadriz** [nvim-lualine/lualine.nvim#1041]
### Self Checks

- [ x] I'm using the latest lualine.
- [ x] I didn't find the issue in existing issues or PRs.

### How to reproduce the problem
1. `git init test | cd test`
2. Open neovim with minimal config

### Expected behaviour
Lualine sets background and foreground color of diagnostic section to selected highlight group 

Getting something like this:
![Screenshot from 2023-05-08 22-50-18](https://user-images.githubusercontent.com/67771985/236973738-d7a9baa5-a942-48fb-87db-ac3eca8381b3.png)

### Actual behaviour
Error:
```
Failed to run `config` for lualine.nvim
...m/plugins/lualine.nvim/lua/lualine/utils/color_utils.lua:286: attempt to perform arithmetic on a nil value
# stacktrace:
  - minimal.lua:28 _in_ **config**
  - minimal.lua:65
```

### Minimal config to reproduce the issue
```lua
local temp_dir = vim.loop.os_getenv "TEMP" or "/tmp"
local install_dir = temp_dir .. "/lazy-nvim"

-- set stdpaths to use "/tmp/lazy-nvim"
for _, name in ipairs { "config", "data", "state", "cache" } do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = install_dir .. "/" .. name
end

-- bootstrap lazy
local lazypath = install_dir .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system {
        "git",
        "clone",
        "--filter=blob:none",
        "--single-branch",
        "https://github.com/folke/lazy.nvim.git",
        lazypath,
    }
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
    {
        "nvim-lualine/lualine.nvim",
        config = function()
            require("lualine").setup {
                sections = {
                    lualine_x = {
                        {
                            "diagnostics",
                            sections = { "warn" },
                            symbols = { warn = "" },
                            diagnostics_color = {
                                warn = {
                                    bg = "DiagnosticWarn",
                                    fg = "lualine_c_normal",
                                },
                            },
                        },
                    },
                },
            }
        end,
    },
}

vim.lsp.start {
    name = "lua_ls",
    cmd = { "lua-language-server" },
    root_dir = vim.fs.dirname(vim.fs.find({
        ".git",
    }, { upward = true })[1]),
    settings = { Lua = { telemetry = { enable = false } } },
}

require("lazy").setup(plugins, {
    root = install_dir .. "/plugins",
})

vim.opt.termguicolors = true
```

### Additional information

The minimal config installs everything in `/tmp`, nothing will be left out in your system and it will not mess up with your config in any way, I'm also setting up `lua_ls` without plugins using buildin `vim.lsp` so you can open a file with diagnostics and test if the colors are being setup correctly (you need `lua-language-server` in your path), that's way I included `git init test | cd test` in steps to reproduce so the server can detect a root dir and start. 

The minimal config has errors from diagnostic so open with `nvim -u minimal.lua minimal.lua` 

Basically I'm trying to display my diagnostics like this:
![Screenshot from 2023-05-08 22-39-04](https://user-images.githubusercontent.com/67771985/236975154-c377450a-31c2-40ae-856b-e11caf141b4b.png)

Just a number with the background color of the severity, and the color should be the one of the highlight group, this way it changes depending on the current theme.

I can hardcoded with colors instead of highlight groups and it works:

```lua
diagnostics_color = {
	warn = {
    	bg = "Orange",
        fg = "Black",
	},
}
```
![Screenshot from 2023-05-08 22-50-18](https://user-images.githubusercontent.com/67771985/236973738-d7a9baa5-a942-48fb-87db-ac3eca8381b3.png)

But as I said, I want the background and foreground color to change according to theme.
